### ISSUE TITLE
generic pyFAI applcation

### ISSUE THREAD

**Issue by kif** [kif/pyFAI#184]
It would call calib, integrate ... under the hood. 
This would make single-binary distribution much easier (i.e on windows & macosx).

**Comment by toqduj** [comment]
Yes please. Still having trouble building PyFAI on my combination of Mac OS X and Enthought Python (or "Canopy" or whatever they are now calling it). Anything that would bring the program closer to having binaries distributed would be welcome. 

Alternatively, can we have a package list required for successful build on OS X?

**Comment by kif** [comment]
On Thu, 15 Oct 2015 00:49:58 -0700
"Brian R. Pauw" <notifications@github.com> wrote:

> Yes please. Still having trouble building PyFAI on my combination of Mac OS X and Enthought Python (or "Canopy" or whatever they are now calling it). Anything that would bring the program closer to having binaries distributed would be welcome. 
> 
> Alternatively, can we have a package list required for successful build on OS X?

Hi Brian,

Great to hear from you :)
I had pyFAI running smoothly on my mac until I upgraded to El-captain :) now everything looks broken.
The issue with canopy is probably related to the compiler and/or to the option for building python itself.

For installing on macosx, I just did:

It looks like osx provides now python 2.7.10 in /usr/bin with pip2 in /usr/local/bin. 
So I did:
Install Xcode7 from the AppStore (it is free, while heavy !).
Then on the python side:

  sudo pip install numpy
  sudo pip install cython
  sudo pip install h5py
  sudo pip install pyopencl
  sudo pip install scipy 
  sudo pip install matplotlib

To install pyFAI:
  rm -rf build
  python setup.py build --no-openmp
  python setup.py test #needs a network connection, it also spotted 2 new bugs :)
  sudo pip install .
 
  pip wheel . #to build a wheel for pyFAI redistributable on other similar OSX computers

Note Cython is a required dependency on OSX as their native compiler is
not OpenMP compliant and the cython code to be generated is hence
different from the windows/linux one.

Cheers,

-- 
Jérôme Kieffer
tel +33 476 882 445

**Comment by kif** [comment]
About OpenMP compliance of Xcode ...
https://forums.developer.apple.com/thread/18207
I know it is not very constructive, but I have the feeling  of loosing a lot of time with Apple's political point of view.

**Comment by vallsv** [comment]
setuptools make the thing easier now. Is that still something to aim?

**Comment by kif** [comment]
On Tue, 06 Feb 2018 12:10:42 +0000 (UTC)
Valentin Valls <notifications@github.com> wrote:

> setuptools make the thing easier now. Is that still something to aim?

not in the near future.

-- 
Jérôme Kieffer
tel +33 476 882 445
